core/app/models/spree/order.rb

```diff
@@ -721,6 +721,15 @@ def validate_payments_attributes(attributes)
       end
     end
 
+    # Returns whether have been added some {#line_items}
+    #
+    # It considers both persisted and in-memory line items.
+    #
+    # @return [Boolean]
+    def items?
```

Comment:
mmm, I'm not 100% convinced by this method. It's used only once in this PR, and if I look in the codebase for `line_items.any?` I can find only occurrences with a block, which are not replaceable with this implementation.

Reply:
Yeah, it doesn't strike me as a particularly valuable abstraction over just calling `line_items.any?`, even if the functionality is useful.

Reply:
Good point. We can also modify the method to accept the block and replace the other occurrences while keeping Demeter happy :slightly_smiling_face: WDYT?

Reply:
I would be more convinced, but still not 100%, for these reasons:
* the current name doesn't directly map to what it aliases, so it increases the cognitive distance (but this can be easily fixed by changing to `line_items_any?`);
* when we add this method, then why not adding a similar one for any occurrence of `whatever_association#any?` we have in the codebase? If it is good this time, it then should be always good... but I think this would add mostly overhead;
* some people don't agree we should keep Demeter happy all the time, for example https://www.tedinski.com/2018/12/18/the-law-of-demeter.html. In this case, I think that the biggest value for this change would be when adding clarity/removing a lot of duplication, but we don't have such thing now, since:
  * there are only 2 occurrences (3 with your change) in the application code, plus another 2 in specs;
  * it seems to be perfectly fine in the Rails ecosystem to call `#association_name.any?` (with or without the block) so I think it's very unlikely that other people (contributors, developers) will start using that method, even when aware of its existence;

By the way, I think it would be totally fine if ActiveRecord added something similar, as it would be at the API level and everybody would be encouraged to know about it and start using it. But I see little value in having this single custom method in this Solidus class.

Reply:
I agree that the Law of Demeter shouldn't be seen as carved in stone. But, in general, I don't like to ask for data and perform logic afterward, as objects should keep data and behavior together. I also understand this is neither carved in stone, but I think that most of the cases when this doesn't apply reveals you have a problem with your design. It's not the case here, or at least we have little control over it, as they are standard Rails associations.

But I also understand that it can also lead to bloated interfaces. With AR having so many public methods, if we did that for every method, then we would land on that scenario. In my view, in this case, asking whether an order has line items should be packed with the order itself, but I agree it's a moot point, so I've changed the code along with your suggestion.